docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,212 @@
+
+## Tradeoffs
+
+There are some tradeoffs to consider when choosing a namespace ID size.
+
+### NMT Node Size
+
+The namespace ID is prefixed to each NMT data leaf. Two namespace IDs are prefixed to each NMT non-leaf hash. Therefore, the nodes of an NMT will be larger based on the namespace ID size. Assuming shares are 512 bytes:
+
+Namespace ID size (bytes) | NMT data leaf size (bytes) | NMT inner node size (bytes)
+--------------------------|----------------------------|----------------------------
+8                         | 8 + 512 = 520              | 2*8 + 32 = 48
+16                        | 16 + 512 = 528             | 2*16 + 32 = 64
+20                        | 20 + 512 = 532             | 2*20 + 32 = 72
+32                        | 32 + 512 = 544             | 2*32 + 32 = 96
+
+### NMT Proof Size
+
+Increasing the size of NMT nodes will increase the size of the NMT proof. Assuming shares are 512 bytes, square size is 128, and the NMT proof is for a single leaf:
+
+Namespace ID size (bytes) | Unencoded NMT proof size (bytes) | Protobuf encoded NMT proof size (bytes) | Protobuf encoded NMT proof with [gzip](https://pkg.go.dev/compress/gzip) (bytes)
+--------------------------|----------------------------------|-----------------------------------------|---------------------------------------------------------------------------------
+8                         | 336                              | 354                                     | 382
+16                        | 448                              | 466                                     | 408
+20                        | 504                              | 522                                     | 466
+32                        | 672                              | 690                                     | 630
+
+### Blob Inclusion Proof Size
+
+Blob inclusion proofs haven't yet been implemented so this proposal can't precisely determine the impact on blob inclusion proofs. A naive implementation of blob inclusion proofs may return NMT proofs for all shares that a blob occupies, in other words one NMT proof per row that a blob spans. Assuming shares are 512 bytes, square size is 128, and a blob is less than 128 shares, a blob would occupy a maximum of 2 rows. Therefore, the namespace ID size's impact on blob inclusion proofs would be approximately 2 * the impact on NMT proofs. A [blob size independent inclusion proof](https://github.com/celestiaorg/celestia-app/blob/6d27b78aa64a749a808e84ea682352b8b551fbd7/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md?plain=1#L19) is likely smaller than this naive implementation because it depends on the number of shares that a PFB transaction spans (likely significantly fewer than 2 rows).
+
+### Share Size
+
+Another tradeoff to consider is the size of the namespace ID in the share. Since a share is a fixed 512 bytes, a share's capacity for blob data decreases as the namespace ID increases.
```

Comment:
[Sharing thoughts] The total size of a blob is also affected by the ID size. For example, if a 32-byte ID is used instead of an 8-byte ID, the total size of a blob is reduced by 128 * (32-8) (128 refers to the max number of shares per blob as mentioned earlier).

Reply:
Yea good point. Will add a section for it.

Reply:
Resolved by 70fb1308288b7bcb2a4bfbe152496b16aea89a98